indico/htdocs/js/indico/modules/registration/form/tpls/fields/accommodation.tpl.html

```diff
@@ -123,4 +79,56 @@
             value="{{ accommodation.choice }}"/>
         <input type="hidden" name="{{ field.htmlName }}" value="{{ userdata[fieldName] }}">
     </ng-form>
+    <table cellspacing="0">
+        <tr>
+            <td align="left">
+                <span>{{ 'Arrival date' | i18n }}</span>
+                <span ng-if="field.isRequired" class="regFormMandatoryField">*</span>
+            </td>
+            <td align="left">
+                <ng-form name="nestedFormArrival">
+                    <select required
+                        id="arrivalDate"
+                        ng-model="accommodation.arrivalDate"
+                        ng-class="{hasError: validationStarted && nestedFormArrival.$invalid}"
+                        ng-disabled="(!userdata.manager && billableOptionPayed(userdata)) || !accommodation.choice"
+                        ng-required="field.isRequired">
+                        <option value="" selected>{{ 'Select a date' | i18n }}</option>
+                        <option ng-repeat="date in field.arrivalDates" value="{{ date[0] }}"
+                                ng-selected="userdata[fieldName].arrivalDate == date[0]">
+                            {{ date [1]}}
+                        </option>
+                    </select>
+                    <input type="hidden"
```

Comment:
I think (but test it!) we don't need those hidden fields anymore since the field implementation on the server side ignores the value if the selected option has been paid for.